# Biological-interpretation-packages

After performing genomic alignment and statistical analysis to identify the differentially expressed genes (DEGs) based on the value and adjusted p-value, the next step is to analyze the DEGs in relation to the associated biological pathways. 

We used two packages for in silico analysis: ClusterProfiler, which also facilitates the analysis and interpretation of biological data, utilizing the KEGG database (Kyoto Encyclopedia of Genes and Genomes, version 113.0, source: Kanehisa Laboratories) and the Gene Ontology database (https://www.geneontology.org/, version 3.18.0, sources: Alliance of Genome Resources and Global Biodata Coalition). 

Additionally, we used the Database for Annotation, Visualization, and Integrated Discovery (DAVID - https://davidbioinformatics.nih.gov/), which incorporates various databases, with NCBI and Uniprot being the primary ones. 


To identify the names of differentially expressed genes, you can use the Biomart platform (https://www.ensembl.org/biomart/martview/0aec670f1cb496f045d89910c08c54b0), which facilitates the retrieval of gene names of interest.
